Source/Core/Runtime/RestrictiveEnvironment/DefaultStepLockHandling.cs

```diff
@@ -47,7 +47,7 @@ public override void Lock(IStepData data, IEnumerable<LockablePropertyData> manu
                 IStepData nextStepData = GetNextStep(completedTransition);
                 IEnumerable<LockablePropertyData> nextStepProperties = PropertyReflectionHelper.ExtractLockablePropertiesFromStep(nextStepData);
 
-                if (nextStepData != null && nextStepData is ILockableStepData lockableStepData)
+                if (nextStepData is ILockableStepData lockableStepData)
                 {
                     IEnumerable<LockablePropertyData> toUnlock = lockableStepData.ToUnlock.Select(reference => new LockablePropertyData(reference.GetProperty()));
```

Comment:
This line has not changed, I assume it can still give a NRE on reference.GetProperty()?